### PR TITLE
Docusaurus import

### PR DESCRIPTION
**Pull request by NatKarmios**
This brings over old docs from Docusaurus, along with a new theme, some style tweaks and the Gillian logo.

Also some Lifter documentation for good measure.

### CODE REVIEW

**Review by giltho — DISMISSED** [review]

File: sphinx/publications/gillian-part-1-diff.rst [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+The Memory Interfaces
+---------------------
+
+Here is how Memory models are defined in the paper:
+
+    **Definition** *(Concrete Memory Model)*: A concrete memory model, :math:`M \in \mathbb{M}`, is a triple :math:`\langle |M|, A, \underline{\mathsf{ea}}\rangle`, consisting of a set of concrete memories, :math:`|M| \ni \mu`, a set of actions :math:`A \ni \alpha`, and the action execution function :math:`\underline{\mathsf{ea}} : A \rightarrow |M| \rightarrow \mathcal{V} \rightarrow \wp(|M| \times \mathcal{V})`, pretty-printed :math:`\mu.\alpha(v) \rightsquigarrow (\mu', v)`.
+
+    **Definition** *(Symbolic Memory Model)*: A symbolic memory model, :math:`\hat M \in \mathbb{M}`, is a triple :math:`\langle |\hat M|, A, \hat{\underline{\mathsf{ea}}}\rangle`, consisting of a set of symbolic memories, :math:`|\hat M| \ni \hat \mu`, a set of actions :math:`A \ni \alpha`, and the action execution function :math:`\underline{\mathsf{ea}} : A \rightarrow |M| \rightarrow \mathcal{V} \rightarrow \wp(|M| \times \mathcal{V})$, pretty-printed $\mu.\alpha(v) \rightsquigarrow (\mu', v)`, pretty-printed :math:`\hat \mu.\alpha(\hat e) \rightarrow (\mu', \hat e', \pi ')`.
```
[the remark this answers is not in the document]
Comment (giltho) [reply sphinx/publications/gillian-part-1-diff.rst:137]: also, later down the page
image

[the remark this answers is not in the document]
Comment (giltho) [reply sphinx/publications/gillian-part-1-diff.rst:137]: Gah finding more
image

File: GillianCore/debugging/lifter/gil_fallback_lifter.mli [hunk 2]
```diff
@@ -1,2 +1,12 @@
+(** A {!Lifter} implementation that acts as a proxy to another lifter
+  ([TLLifter]), while also keeping a {!Gil_lifter} updated alongside.
+  
+  [TLLifter] can access the GIL lifter and its state via
+  {!Gil_lifter_with_state}.
+  
+  Note that if the GIL lifter gives anything other than [Stop] when handling
+  a command, it is considered skipped, and [TLLifter] won't be asked to
+  handle it. *)
+
```
[the remark this answers is not in the document]
Comment (NatKarmios) [reply GillianCore/debugging/lifter/gil_fallback_lifter.mli:10]: Good idea.
#201

File: sphinx/publications/gillian-part-1-diff.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (NatKarmios) [reply sphinx/publications/gillian-part-1-diff.rst]: Good spots, thanks.
Fixed in 97a34044e61e61797a577959bde4a9af17372da1

**Review by giltho — APPROVED** [review]

**Pull request merged**